src/context/ChannelContext.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { PropsWithChildren, useContext } from 'react';
+
+import type {
+  Data as EmojiMartData,
+  NimbleEmojiIndex,
+  NimbleEmojiProps,
+  NimblePickerProps,
+} from 'emoji-mart';
+import type {
+  Attachment,
+  Channel,
+  Message,
+  MessageResponse,
+  Mute,
+  ChannelState as StreamChannelState,
+  StreamChat,
+  UpdateMessageAPIResponse,
+  UserResponse,
+} from 'stream-chat';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../types/types';
+
+export type CommonEmoji = {
+  custom: boolean;
+  emoticons: [];
+  short_names: [];
+};
+
+export type EmojiSetDef = {
+  custom: boolean;
+  emoticons: [];
+  short_names: [];
+};
+
+export type MinimalEmoji = CommonEmoji &
+  EmojiSetDef & {
+    colons: string;
+    id: string;
+    name: string;
+    sheet_x: number;
+    sheet_y: number;
+  };
+
+export type EmojiConfig = {
+  commonEmoji: CommonEmoji;
+  defaultMinimalEmojis: MinimalEmoji[];
+  Emoji: React.ComponentType<NimbleEmojiProps>;
+  emojiData: EmojiMartData;
+  EmojiIndex: NimbleEmojiIndex;
+  EmojiPicker: React.ComponentType<NimblePickerProps>;
+  emojiSetDef: {
+    sheetColumns: number;
+    sheetRows: number;
+    sheetSize: number;
+    size: number;
+    spriteUrl: string;
+  };
+};
+
+export type MessageToSend<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  attachments?: Array<
+    | Attachment<At>
+    | {
+        fallback: string;
+        image_url: string | undefined;
+        type: string;
+      }
+    | {
+        asset_url: string;
+        file_size: number;
+        mime_type: string;
+        title: string;
+        type: string;
+      }
+  >;
```

Comment:
what's up with this? should these be thrown on the default attachment type as optionals instead of done like this?

Reply:
looks like only ones not on the Attachment type are mime_type and file_size, will just add these as optional and extend the default type